Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoMeraki.yaml

```diff
@@ -37,78 +37,609 @@ ParserParams:
+      | extend SrcMacAddr = case(
+                            isnotempty(SrcMacAddr),
+                            SrcMacAddr,
+                            LogType == "airmarshal_events",
+                            src, 
+                            coalesce(client_mac, mac, shost)
+                        )
+      | extend SrcMacAddr =  iff(
+                            SrcMacAddr has '"',
+                            replace_string(tostring(SrcMacAddr), '"', ''),
+                            replace_string(tostring(SrcMacAddr), "'", "")
+                        )
+      | extend DstMacAddr = iff(LogType == "airmarshal_events", dst, dhost)
+      | extend DstMacAddr =  iff(
+                            DstMacAddr has '"',
+                            replace_string(tostring(DstMacAddr), '"', ''),
+                            replace_string(tostring(DstMacAddr), "'", "")
                         )
-          | where ASimMatchingIpAddr != "No match"
-          | extend temp_isMatchSrcHostname= SrcHostname has_any (hostname_has_any),
-                   temp_isMatchDstHostname = DstHostname has_any (hostname_has_any)
-          | extend ASimMatchingHostname = case(
-                            array_length(hostname_has_any) == 0 , "-", 
-                            (temp_isMatchSrcHostname and temp_isMatchDstHostname), "Both" , 
-                            temp_isMatchSrcHostname, "SrcHostname", 
-                            temp_isMatchDstHostname, "DstHostname", 
-                            "No match"
+      | extend SrcIpAddr = iff(SrcIpAddr == SrcMacAddr, "", SrcIpAddr)
+      | extend DstIpAddr = iff(DstIpAddr == DstMacAddr, "", DstIpAddr)
+      | extend DvcMacAddr = column_ifexists("wired_mac", "")
+      | extend DvcMacAddr = iff(
+                            DvcMacAddr has '"',
+                            replace_string(tostring(DvcMacAddr), '"', ''),
+                            replace_string(tostring(DvcMacAddr), "'", "")
                         )
-          | project-away temp*
-          | where ASimMatchingHostname != "No match"
-          | extend  
-              EventResult = iff(DvcAction == "Deny", "Failure", "Success"),
-              EventResultDetails = "",
-              NetworkDuration = toint(NetworkDuration)
-          | extend 
-              InnerVlanId = SrcVlanId,
-              OuterVlanId = DstVlanId,
-              SessionId = NetworkSessionId,
-              Hostname = DstHostname,
-              IpAddr = SrcIpAddr,
-              Duration = NetworkDuration, 
-              Dst = DstIpAddr,
-              Src = SrcIpAddr,
-              User = DstUsername
-        | project-away Url*, Http*, File*, Rule*, UserAgent, Hash
-      };
-    NWParser (starttime, endtime, srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix, dstportnumber, hostname_has_any, dvcaction, eventresult, disabled)
\ No newline at end of file
+      | where  (isnull(dstportnumber) or dstportnumber == DstPortNumber)
```

Comment:
Again, there is a scope of prefiltering.

Reply:
Done